lib/Sema/CSApply.cpp

```diff
@@ -604,7 +606,10 @@ namespace {
       if (isa<AbstractFunctionDecl>(member)) {
         // For functions, close the existential once the function
         // has been fully applied.
+        assert(member->getDeclContext()->isTypeContext());
         return 2;
+      } else if (auto *elt = dyn_cast<EnumElementDecl>(member)) {
+        return elt->getParameterList() ? 2 : 1;
```

Comment:
Can the brains of this function just be `ValueDecl::getNumCurryLevels()`?  The assertions are still good to keep.

Reply:
Yup! Thanks.